enhancements/sig-architecture/58-addon-configuration/README.md

```diff
@@ -0,0 +1,606 @@
+    // addOnConfiguration is a reference to configuration information for the add-on.
+    // This resource is use to locate the configuration resource for the add-on.
+    // +optional
+    AddOnConfiguration ConfigCoordinates `json:"addOnConfiguration",omitempty`
+
+    // configReference is a reference to the add-on configuration.
+    // +optional
+    ConfigReference ConfigReference `json:"configReference,omitempty"`
+}
+
+// ConfigReference is a reference to the current add-on configuration.
+// This resource is used to locate the configuration resource for the current add-on.
+type ConfigReference struct {
+    // configGVR represents the add-on configuration GroupVersionResource for current add-on.
+    // This field is synced from ClusterManagementAddOn configGVR field.
+    ConfigGVR ConfigGVR `json:"configGVR"`
+
+    // config represents the add-on configuration namespace and name for current add-on.
+    // This field is synced from ClusterManagementAddOn defaultConfig and ManagedClusterAddOn config fields.
+    // If both them are specified, the ManagedClusterAddOn config will overwrite the ClusterManagementAddOn
+    // defaultConfig.
+    Config ConfigReferent `json:"config"`
+
+    // lastObservedGeneration is the observed generation of the add-on configuration.
+    LastObservedGeneration int64 `json:"lastObservedGeneration"`
+}
+```
+
+The add-on framework will watches the `ManagedClusterAddOn` to get the add-on configuration reference from the 
+`ManagedClusterAddOn` status.
+
+If the `ManagedClusterAddOn` has the configuration reference, the add-on framework use the `configGVR` and `config` to
+locate the configuration resource and watch the configuration resource, once the configuration resource
+changes, the add-on famework updates the `ManagedClusterAddOn` status (`lastObservedGeneration`), this will trigger
+the [`Manifests`](https://github.com/open-cluster-management-io/addon-framework/blob/main/pkg/agent/inteface.go#L31)
+interface which add-on implement.
+
+As a specific case, we want to introduce a `AddOnDeploymentConfig` API in cluster scope to configure the add-on
+deployment.
+
+This API is optional for add-on developers, add-on developers can choose to use this API to configure their add-on
+deployment or use their own API for configuration.
+
+```golang
+// AddOnDeploymentConfig represents a deployment configuration for an add-on.
+// AddOnDeploymentConfig is a cluster-scoped resource.
+type AddOnDeploymentConfig struct {
+    metav1.TypeMeta   `json:",inline"`
+    metav1.ObjectMeta `json:"metadata,omitempty"`
+
+    // spec represents a desired configuration for an add-on.
+    // +required
+    Spec AddOnDeploymentConfigSpec `json:"spec"`
+}
+
+type AddOnDeploymentConfigSpec struct {
+    // CustomizedVariables is a list of name-value variables for the current add-on deployment.
+    // The add-on implementation can use these variables to render the add-on deployment.
```

Comment:
I worry about this being used in lieu of defining a schema.  Is there a way to describe what variables are available?

Reply:
sometimes, we need configure some env variables or other fields for deployment, e.g. the http proxy, the imagePullPolicy, etc. we want  to use this to configure them

Reply:
@deads2k do you have any suggestion for this part? is it better to use a generic configuration (like current name-value variables) or use a specific configuration, we define the image, imagePullPolicy and http proxy in this spec?

Reply:
I gave this some thought over the weekend and I cannot think of a compact way to avoid this problem.  I think we could limit the problems with free-form fields by suggesting that the managedclusteraddon report a condition if the CustomizedVariables included an unrecognized name or if the rendering fails.

Reply:
thank you, I think report the condition is a good suggestion, I will add this in the enhancement

Reply:
I add a new condition `Configured` to the managedclusteraddon, I think we can use this condition to report the condition when we configure an addon with its configuration